File: 1998/1998.go
func gcdSortOfAnArray(root *TreeNode) int {
	if root == nil {
		return 0
	}
	l := gcdSortOfAnArray(root.Left)
	r := gcdSortOfAnArray(root.Right)
	if l > r {
		return 1 + l
	}
	return 1 + r
}

File: 1998/1998.js
/**
 * @param {TreeNode} root
 * @return {number}
 */
var gcdSortOfAnArray = function(root) {
    if (!root) return 0;
    return 1 + Math.max(gcdSortOfAnArray(root.left), gcdSortOfAnArray(root.right));
};